src/createHeadlessForm.js

```diff
@@ -323,9 +324,10 @@ export function createHeadlessForm(jsonSchema, customConfig = {}) {
   };
 
   try {
-    const fields = getFieldsFromJSONSchema(jsonSchema, config);
+    const validations = createValidationChecker(jsonSchema);
```

Comment:
Update. I'm already on Part 7 and faced the same thought again multiple times across the review. Having parameters called `validations` is misleading because it's not a normal one. I'd call this either `logicValidations` (specific) or `advancedValidations` (more generic if one day we drop jsonlogic), or simply `logic` as it's smaller yet unique.

Reply:
i like `logic`, renaming to that!